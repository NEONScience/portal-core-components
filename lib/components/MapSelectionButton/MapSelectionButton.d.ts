export default WrappedMapSelectionButton;
declare const WrappedMapSelectionButton: any;
