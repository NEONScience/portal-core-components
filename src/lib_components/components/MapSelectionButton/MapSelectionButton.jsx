import React, { Suspense, useState } from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import Tooltip from '@material-ui/core/Tooltip';
import Typography from '@material-ui/core/Typography';

import CloseIcon from '@material-ui/icons/Close';
import DoneIcon from '@material-ui/icons/Done';
import GlobeIcon from '@material-ui/icons/Language';

import Theme from '../Theme/Theme';
import {
  FEATURE_TYPES,
  SITE_MAP_PROP_TYPES,
  getDefaultState,
} from '../SiteMap/SiteMapUtils';

const SiteMap = React.lazy(() => import('../SiteMap/SiteMap'));

const useStyles = makeStyles((theme) => ({
  appBar: {
    position: 'relative',
    paddingRight: '0px !important',
  },
  appBarTitle: {
    marginLeft: theme.spacing(2),
    flex: 1,
  },
  toolbar: {
    padding: '0px 10px 0px 14px',
  },
  toolbarClose: {
    '&:hover': {
      backgroundColor: '#ffffff33',
    },
    '&:not(:hover):not(:focus)': {
      backgroundColor: '#ffffff11',
      border: `1px solid ${theme.palette.primary.main}11`,
    },
  },
  buttonIcon: {
    marginLeft: theme.spacing(1),
  },
  suspenseFallback: {
    width: '100%',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.palette.grey[200],
    fontSize: '1.5em',
  },
}));

const MapSelectionButton = (props) => {
  const {
    label,
    icon,
    dialogTitle: dialogTitleProp,
    buttonProps,
    siteMapProps,
    tooltipProps,
    selection: selectionProp,
    validItems,
    selectedItems,
    selectionLimit,
    onSave,
  } = props;

  const classes = useStyles(Theme);
  const [dialogOpen, setDialogOpen] = useState(false);
  const [dialogEntered, setDialogEntered] = useState(false);
  const [selection, setSelection] = useState(getDefaultState().selection);

  let unit = '';
  let units = '';
  if (selectionProp) {
    unit = FEATURE_TYPES[selectionProp].unit || '';
    units = FEATURE_TYPES[selectionProp].units || '';
  }

  let dialogTitle = `Select ${units}`;
  if (typeof selectionLimit === 'number') {
    dialogTitle = selectionLimit === 1
      ? `Select a ${unit}`
      : `Select ${selectionLimit.toString()} ${units}`;
  }
  if (Array.isArray(selectionLimit)) {
    const { 0: min, 1: max } = selectionLimit;
    dialogTitle = min === 1
      ? `Select up to ${max.toString()} ${units}`
      : `Select ${min.toString()}-${max.toString()} ${units}`;
  }

  const saveTooltipProps = selection.valid ? {} : {
    disableFocusListener: true,
    disableHoverListener: true,
    disableTouchListener: true,
  };

  const aspectRatio = (window.innerHeight - 64) / window.innerWidth;

  const embedProps = {
    fullscreen: true,
    selection: selectionProp,
    aspectRatio,
    validItems,
    selectedItems,
    selectionLimit,
    onSelectionChange: setSelection,
  };
  const finalEmbedProps = siteMapProps ? {
    ...siteMapProps,
    ...embedProps,
  } : embedProps;

  const suspenseFallback = (
    <div className={classes.suspenseFallback}>
      Loading Map...
    </div>
  );

  return (
    <div>
      <Tooltip
        title={`${dialogTitle} using the observatory map`}
        aria-label={`${dialogTitle} using the observatory map`}
        {...tooltipProps}
      >
        <Button
          color="primary"
          variant="contained"
          data-selenium="map-selection-button"
          startIcon={icon ? <GlobeIcon /> : null}
          {...buttonProps}
          onClick={() => setDialogOpen(true)}
        >
          {label}
        </Button>
      </Tooltip>
      <Dialog
        fullScreen
        open={dialogOpen}
        onClose={() => setDialogOpen(false)}
        TransitionProps={{
          onEntered: () => setDialogEntered(true),
        }}
      >
        <AppBar color="secondary" className={classes.appBar}>
          <Toolbar className={classes.toolbar}>
            <Tooltip
              title="Exit selection without saving"
              aria-label="Exit selection without saving"
            >
              <IconButton
                edge="start"
                color="primary"
                variant="contained"
                aria-label="close"
                size="small"
                className={classes.toolbarClose}
                onClick={() => setDialogOpen(false)}
              >
                <CloseIcon />
              </IconButton>
            </Tooltip>
            <Typography variant="h5" className={classes.appBarTitle}>
              {dialogTitleProp || dialogTitle}
            </Typography>
            <Tooltip
              title="Complete selection and return"
              aria-label="Complete selection and return"
              {...saveTooltipProps}
            >
              <div>
                <Button
                  color="primary"
                  variant="contained"
                  onClick={() => { onSave(selection.set); setDialogOpen(false); }}
                  disabled={!selection.valid}
                >
                  Save
                  <DoneIcon className={classes.buttonIcon} />
                </Button>
              </div>
            </Tooltip>
          </Toolbar>
        </AppBar>
        {!dialogEntered ? null : (
          <Suspense fallback={suspenseFallback}>
            <SiteMap {...finalEmbedProps} />
          </Suspense>
        )}
      </Dialog>
    </div>
  );
};

MapSelectionButton.propTypes = {
  label: PropTypes.string,
  icon: PropTypes.bool,
  dialogTitle: PropTypes.string,
  buttonProps: PropTypes.shape(Button.propTypes),
  siteMapProps: PropTypes.shape(SITE_MAP_PROP_TYPES),
  tooltipProps: PropTypes.shape(Tooltip.propTypes),
  selection: SITE_MAP_PROP_TYPES.selection.isRequired,
  validItems: SITE_MAP_PROP_TYPES.validItems,
  selectedItems: SITE_MAP_PROP_TYPES.selectedItems,
  selectionLimit: SITE_MAP_PROP_TYPES.selectionLimit,
  onSave: PropTypes.func,
};

MapSelectionButton.defaultProps = {
  label: 'Map',
  icon: true,
  dialogTitle: null,
  buttonProps: {},
  siteMapProps: null,
  tooltipProps: {},
  validItems: null,
  selectedItems: [],
  selectionLimit: null,
  onSave: () => {},
};

const WrappedMapSelectionButton = Theme.getWrappedComponent(MapSelectionButton);

export default WrappedMapSelectionButton;
