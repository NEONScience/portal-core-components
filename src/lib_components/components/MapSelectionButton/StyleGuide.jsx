/* eslint-disable react/jsx-one-expression-per-line, jsx-a11y/anchor-is-valid, react/no-unescaped-entities, max-len */
import React, { useState } from 'react';

import { makeStyles } from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';

import DocBlock from '../../../components/DocBlock';
import CodeBlock from '../../../components/CodeBlock';
import ExampleBlock from '../../../components/ExampleBlock';
import PropsTable from '../../../components/PropsTable';

import { FEATURE_TYPES } from '../SiteMap/SiteMapUtils';

import MapSelectionButton from './MapSelectionButton';

import Theme from '../Theme/Theme';

const useStyles = makeStyles((theme) => ({
  divider: {
    margin: theme.spacing(3, 0),
  },
}));

const buttonApiLink = <Link href="https://material-ui.com/api/button/">Material UI Button</Link>;
const tooltipApiLink = <Link href="https://material-ui.com/api/tooltip/">Material UI Tooltip</Link>;

const propRows = [
  {
    name: 'buttonProps',
    type: 'object',
    default: 'null',
    description: (
      <p>
        An object of {buttonApiLink} props to apply to the button. Applied after all hard-coded
        props except for <tt>onClick</tt> (meaning all other props can be overridden).
      </p>
    ),
  },
  {
    name: 'dialogTitle',
    type: 'string',
    default: 'null',
    description: (
      <p>
        The title that will appear at the top of the dialog. If not defined a generic title will be
        produced taking into account the value of <tt>selectionLimit</tt>.
      </p>
    ),
  },
  {
    name: 'icon',
    type: 'boolean',
    default: 'true',
    description: (
      <p>
        Whether to include the globe icon for the button.
      </p>
    ),
  },
  {
    name: 'label',
    type: 'string',
    default: '"map"',
    description: (
      <p>
        The label that will appear on the button.
      </p>
    ),
  },
  {
    name: 'onSave',
    type: 'function',
    default: 'null',
    description: (
      <p>
        A function that fires when the selection is committed. Does not fire on every change, only
        when the workflow is complete and the user clicks Save on a valid selection. It should take
        one argument representing the updated selection as a Set (not an array).
      </p>
    ),
  },
  {
    name: 'selectedItems',
    type: 'array of strings',
    default: '[]',
    examples: (
      <div>
        <tt>['ABBY', 'WREF', 'CPER']</tt>
        <br />
        <tt>['JORN']</tt>
      </div>
    ),
    description: (
      <p>
        Array of strings representing IDs of selectable items to show already selected when the
        SiteMap initializes. For example, if <tt>selection</tt> is <tt>"SITES"</tt> then this prop
        should contain siteCode strings.
      </p>
    ),
  },
  {
    name: 'selection',
    type: (
      <div>
        string, one of:
        {Object.keys(FEATURE_TYPES).filter((k) => FEATURE_TYPES[k].selectable).map((k) => (
          <div key={k} style={{ marginTop: '8px' }}>
            <tt>{`"${k}"`}</tt>
          </div>
        ))}
      </div>
    ),
    default: 'n/a (required)',
    description: (
      <p>
        <b>Required</b> string representing the feature type to be selectable in the map. All map
        features are grouped into types, and certain types are selectable.
      </p>
    ),
  },
  {
    name: 'selectionLimit',
    type: 'number or array of exactly two numbers',
    default: 'null',
    examples: (
      <div>
        <tt>3</tt>
        <br />
        <tt>[1, 5]</tt>
      </div>
    ),
    description: (
      <>
        <p>
          When selection is active and this prop is null there are no limits on what constitutes a
          valid selection (as long as at least one selectable item is selected). When not null this
          prop enforces limits on that validity.
        </p>
        <p>
          A numeric value will set a discrete selection limit (e.g. a value of <tt>3</tt> means
          that <i>exactly</i> three items must be selected to be valid). An array of two values
          sets a range (e.g. <tt>[1, 5]</tt> will consider a selection anywhere from one item up
          to five items to be valid).
        </p>
        <p>
          For all numeric values for this prop only non-zero positive integers are allowed. For an
          array of two numbers the second number must be greater than (and not equal two) the first.
        </p>
      </>
    ),
  },
  {
    name: 'siteMapProps',
    type: 'object',
    default: 'null',
    description: (
      <p>
        An object of props to apply to the <Link href="#SiteMap">Site Map</Link>. Use this to
        control additional settings on the embedded SiteMap such as location, zoom, etc. This object
        is spread onto the embedded Site Map <i>before</i> the other named SiteMap props are
        applied, so any named SiteMap props (<tt>selectionLimit</tt>, <tt>validItems</tt>,
        and <tt>selectionLimit</tt>) will defer to the top-level definition.
      </p>
    ),
  },
  {
    name: 'tooltipProps',
    type: 'object',
    default: 'null',
    description: (
      <p>
        An object of {tooltipApiLink} props to apply to the tooltip surrouning the button. Applied
        after hard-coded props so all props can be overridden.
      </p>
    ),
  },
  {
    name: 'validItems',
    type: 'array of strings',
    default: '[]',
    examples: (
      <div>
        <tt>['ABBY', 'WREF', 'CPER']</tt>
        <br />
        <tt>['JORN']</tt>
      </div>
    ),
    description: (
      <p>
        Array of strings representing IDs of selectable items. If non-empty this will be interpreted
        as a subset of the set of all items for the selectable type that are enabled for selection.
        Any items not represented in <tt>validItems</tt> will appear ghosted and will not be
        selectable.
      </p>
    ),
  },
];

const MyComponent = () => {
  const [selection, setSelection] = useState(new Set());
  return (
    <div>
      <h3>{`Selected Sites (${selection.size})`}</h3>
      <div>
        {!selection.size ? <i>none</i> : [...selection].join(', ')}
      </div>
      <br />
      <MapSelectionButton
        selection="SITES"
        label="Select Sites"
        selectedItems={[...selection]}
        onSave={(newSelection) => { setSelection(newSelection); }}
      />
    </div>
  );
};

export default function StyleGuide() {
  const classes = useStyles(Theme);

  return (
    <>

      <DocBlock>
        A button to launch the <Link href="#SiteMap">Site Map</Link> in a full screen
        selection-mode dialog.
      </DocBlock>
      <CodeBlock>
        {`
import MapSelectionButton from 'portal-core-components/lib/components/MapSelectionButton';
        `}
      </CodeBlock>

      <Typography variant="h4" component="h2" gutterBottom>Props</Typography>

      <DocBlock>
        <PropsTable props={propRows} />
      </DocBlock>

      <Divider className={classes.divider} />
      <Typography variant="h5" component="h3" gutterBottom>Usage</Typography>

      <DocBlock>
        The Map Selection Button takes on many of the selection-related props of the Site Map (which
        are passed directly through to the embedded Site Map component). It is effectively a button
        with an attached dialog to launch the selected workflow. The examples below demonstrate
        different seed props.
      </DocBlock>
      <ExampleBlock>
        <MapSelectionButton
          selection="SITES"
          selectedItems={['RMNP', 'REDB']}
        />
        <MapSelectionButton
          selection="SITES"
          label="Select a site from D07"
          dialogTitle="Select a site from D07"
          selectionLimit={1}
          validItems={['GRSM', 'LECO', 'MLBS', 'ORNL', 'WALK']}
          siteMapProps={{ location: 'D07' }}
          tooltipProps={{
            title: 'Select exactly one site from Domain D07',
            'aria-label': 'Select exactly one site from Domain D07',
          }}
        />
        <MapSelectionButton
          selection="SITES"
          label="Select up to 4 sites"
          dialogTitle="Select up to 4 sites"
          selectionLimit={[1, 4]}
          icon={false}
        />
      </ExampleBlock>
      <CodeBlock>
        {`
<MapSelectionButton
  selection="SITES"
  selectedItems={['RMNP', 'REDB']}
/>

<MapSelectionButton
  selection="SITES"
  label="Select a site from D07"
  dialogTitle="Select a site from D07"
  selectionLimit={1}
  validItems={['GRSM', 'LECO', 'MLBS', 'ORNL', 'WALK']}
  siteMapProps={{ location: 'D07' }}
  tooltipProps={{
    title: 'Select exactly one site from Domain D07',
    'aria-label': 'Select exactly one site from Domain D07',
  }}
/>

<MapSelectionButton
  selection="SITES"
  label="Select up to 4 sites"
  dialogTitle="Select up to 4 sites"
  selectionLimit={[1, 4]}
  icon={false}
/>
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Saving Changes</Typography>

      <DocBlock>
        Unlike the Site Map component the Map Selection Button component supports an <tt>onSave</tt> method.
        This function will only fire when the Save button is clicked, which is only possible with a
        valid selection. This is the primary mechanism of exporting a selection.
      </DocBlock>
      <ExampleBlock>
        <MyComponent />
      </ExampleBlock>
      <CodeBlock>
        {`
import React, { useState } from 'react';
import MapSelectionButton from 'portal-core-components/lib/components/MapSelectionButton';

const MyComponent = () => {
  const [selection, setSelection] = useState(new Set());
  return (
    <div>
      <h3>{\`Selected Sites ($\{selection.size})\`}</h3>
      <div>
        {!selection.size ? <i>none</i> : [...selection].join(', ')}
      </div>
      <br />
      <MapSelectionButton
        selection="SITES"
        label="Select Sites"
        selectedItems={[...selection]}
        onSave={(newSelection) => { setSelection(newSelection); }}
      />
    </div>
  );
};

<MyComponent />
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Selecting States and/or Domains</Typography>

      <DocBlock>
        Passing an alternate feature type of <tt>"STATES"</tt> or <tt>"DOMAINS"</tt> for
        the <tt>selection</tt> prop will put the map into selection mode for that type. As all other
        props pertaining to valid items, existing selection, and selection limits are all
        type-agnostic they should all work as expected.
      </DocBlock>
      <ExampleBlock>
        <MapSelectionButton
          selection="STATES"
          label="Select between 2 and 7 states"
          selectedItems={['CO', 'UT', 'AZ', 'NM']}
          selectionLimit={[2, 7]}
        />
        <MapSelectionButton
          selection="DOMAINS"
          label="Select a domain in the Eastern US"
          validItems={['D01', 'D02', 'D03', 'D04', 'D05', 'D06', 'D07', 'D08']}
        />
      </ExampleBlock>
      <CodeBlock>
        {`
<MapSelectionButton
  selection="STATES"
  label="Select between 2 and 7 states"
  selectedItems={['CO', 'UT', 'AZ', 'NM']}
  selectionLimit={[2, 7]}
/>

<MapSelectionButton
  selection="DOMAINS"
  label="Select a domain in the Eastern US"
  validItems={['D01', 'D02', 'D03', 'D04', 'D05', 'D06', 'D07', 'D08']}
/>
        `}
      </CodeBlock>

    </>
  );
}
