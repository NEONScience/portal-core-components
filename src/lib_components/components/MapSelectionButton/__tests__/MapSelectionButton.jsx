import React from 'react';
import renderer from 'react-test-renderer';

import MockTheme from '../../../../__mocks__/MockTheme';
import MapSelectionButton from '../MapSelectionButton';

describe('MapSelectionButton', () => {
  test('Renders correctly with only a selection', () => {
    const tree = renderer
      .create(<MockTheme><MapSelectionButton selection="SITES" /></MockTheme>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders correctly with numerical selection limit (1)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <MapSelectionButton
            selection="STATES"
            selectionLimit={1}
            label="foo"
          />
        </MockTheme>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders correctly with numerical selection limit (>1)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <MapSelectionButton
            selection="STATES"
            selectionLimit={4}
            siteMapProps={{
              view: 'table',
            }}
          />
        </MockTheme>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders correctly with array selection limit (min 1)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <MapSelectionButton
            selection="SITES"
            selectionLimit={[1, 4]}
            dialogTitle="bar"
          />
        </MockTheme>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders correctly with array selection limit (min >1)', () => {
    const tree = renderer
      .create(
        <MockTheme>
          <MapSelectionButton
            selection="DOMAINS"
            selectionLimit={[2, 5]}
            icon={false}
          />
        </MockTheme>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
