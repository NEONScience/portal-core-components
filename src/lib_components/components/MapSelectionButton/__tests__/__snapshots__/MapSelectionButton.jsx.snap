// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSelectionButton Renders correctly with array selection limit (min >1) 1`] = `
<div>
  <button
    aria-label="Select 2-5 domains using the observatory map"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r0d1wz-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element={true}
    data-selenium="map-selection-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Map
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MapSelectionButton Renders correctly with array selection limit (min 1) 1`] = `
<div>
  <button
    aria-label="Select up to 4 sites using the observatory map"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r0d1wz-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element={true}
    data-selenium="map-selection-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1ei49fk-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LanguageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
    </span>
    Map
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MapSelectionButton Renders correctly with numerical selection limit (>1) 1`] = `
<div>
  <button
    aria-label="Select 4 states using the observatory map"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r0d1wz-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element={true}
    data-selenium="map-selection-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1ei49fk-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LanguageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
    </span>
    Map
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MapSelectionButton Renders correctly with numerical selection limit (1) 1`] = `
<div>
  <button
    aria-label="Select a state using the observatory map"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r0d1wz-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element={true}
    data-selenium="map-selection-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1ei49fk-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LanguageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
    </span>
    foo
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`MapSelectionButton Renders correctly with only a selection 1`] = `
<div>
  <button
    aria-label="Select sites using the observatory map"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-r0d1wz-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element={true}
    data-selenium="map-selection-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1ei49fk-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LanguageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
    </span>
    Map
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
